### PR TITLE
docs: improve docs around trailing slashes in routes

### PR DESCRIPTION
**Pull request by conradthegray**
This PR addresses the confusion around trailing slashes raised in https://github.com/fastify/fastify/issues/3331

Closes https://github.com/fastify/fastify/issues/3331

### CODE REVIEW

**Review by simone-sanfratello — APPROVED** [review]

File: docs/Reference/Server.md [hunk 1]
```diff
@@ -73,15 +73,14 @@ describes the properties available in that options object.
     - [serializerCompiler](#serializercompiler)
     - [schemaErrorFormatter](#schemaerrorformatter)
     - [schemaController](#schemacontroller)
-      - [Ajv 8 as default schema validator](#ajv-8-as-default-schema-validator)
```
Comment (simoneb) [inline docs/Reference/Server.md:76]: why was this removed?

**Comment by simoneb** [comment]
@conradthegray please link the issue using a keyword so when this PR gets merged the issue is automatically closed

File: docs/Reference/Routes.md [hunk 2]
```diff
@@ -6,21 +6,37 @@ The routes methods will configure the endpoints of your application. You have
...
+  - [Async Await](#async-await)
+  - [Promise resolution](#promise-resolution)
+  - [Route Prefixing](#route-prefixing)
+    - [Handling of / route inside prefixed plugins](#handling-of--route-inside-prefixed-plugins)
+  - [Custom Log Level](#custom-log-level)
+  - [Custom Log Serializer](#custom-log-serializer)
+  - [Config](#config)
+  - [Constraints](#constraints)
+    - [Version Constraints](#version-constraints)
+    - [Host Constraints](#host-constraints)
```
[the remark this answers is not in the document]
Comment (rp4rk) [reply docs/Reference/Routes.md:39]: Screenshot 2022-04-22 at 16 11 42

Here's a quick screenshot of the problem

[the remark this answers is not in the document]
Comment (kurtmilam) [reply docs/Reference/Routes.md:39]: It seems this could be fixed by either by removing lines 9-24 or 24-39.

File: docs/Reference/Server.md (hunk 1 above)
Comment (conradthegray) [reply docs/Reference/Server.md:76]: It was an orphan link. [VSCode plugin for Markdown](https://marketplace.visualstudio.com/items?itemName=yzhang.markdown-all-in-one) I have on my machine removed it automatically.

File: docs/Reference/Routes.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kurtmilam) [reply docs/Reference/Routes.md:39]: Look at line 9 and 25 - these seem to be duplicates. Same for 10 and 26, etc., and 17 / 24 seems to be a leftover fragment from one of the links.

[the remark this answers is not in the document]
Comment (kurtmilam) [reply docs/Reference/Routes.md:39]: Thus my suggestion that removing lines 9-24 or lines 24-39 would seem to get rid of the duplicates and the weird leftover fragment on line 24.

[the remark this answers is not in the document]
Comment (conradthegray) [reply docs/Reference/Routes.md]: Done

**Review by Fdawgs — CHANGES_REQUESTED** [review]
New text needs rejigging for conciseness. Will review and provide suggestions next week.

Formatting issues, as already raised by others, also need resolving.

**Review by simoneb — APPROVED** [review]

**Comment by simoneb** [comment]
@Fdawgs thanks. Formatting issues are now solved.

**Review by kurtmilam — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm
